src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -133,8 +140,15 @@ public BucketMetadata deleteEmptyBucket(long bucketId) {
         if (!bucket.getCatalogObjects().isEmpty()) {
             throw new DeleteNonEmptyBucketException();
         }
-        bucketRepository.delete(bucketId);
+        bucketRepository.delete(bucket.getId());
```

Comment:
Shouldn't we delete by name starting from now?

Reply:
Good point. I was thinking about it. But in this case we need to implement the delete method query by ourselves.
In jpa spring repository it's implemented by default the deleting by primary key. I prefer to use tested blocking query from jpa realization.